mcp/mcp-server/src/main/java/software/amazon/smithy/java/mcp/server/HttpMcpProxy.java

```diff
@@ -0,0 +1,190 @@
+import software.amazon.smithy.java.json.JsonSettings;
+import software.amazon.smithy.java.logging.InternalLogger;
+import software.amazon.smithy.java.mcp.model.JsonRpcErrorResponse;
+import software.amazon.smithy.java.mcp.model.JsonRpcRequest;
+import software.amazon.smithy.java.mcp.model.JsonRpcResponse;
+
+public final class HttpMcpProxy extends McpServerProxy {
+    private static final InternalLogger LOG = InternalLogger.getLogger(HttpMcpProxy.class);
+    private static final JsonCodec JSON_CODEC = JsonCodec.builder()
+            .settings(JsonSettings.builder()
+                    .serializeTypeInDocuments(false)
+                    .useJsonName(true)
+                    .build())
+            .build();
+
+    private final ClientTransport<HttpRequest, HttpResponse> transport;
+    private final URI endpoint;
+    private final String name;
+    private final Signer<HttpRequest, ?> signer;
+    private final Duration timeout;
+
+    private HttpMcpProxy(Builder builder) {
+        this.transport = builder.transport != null ? builder.transport : new JavaHttpClientTransport();
+        this.endpoint = URI.create(builder.url);
+        this.name = builder.name != null ? builder.name : sanitizeName(endpoint.getHost());
+        this.signer = builder.signer;
+        this.timeout = builder.timeout != null ? builder.timeout : Duration.ofSeconds(60);
+    }
+
+    private static String sanitizeName(String host) {
+        if (host == null) {
+            return "http-proxy-mcp";
+        }
+        return host.replaceAll("[^a-zA-Z0-9-]", "-");
+    }
+
+    public static class Builder {
+        private String url;
+        private String name;
+        private Signer<HttpRequest, ?> signer;
+        private ClientTransport<HttpRequest, HttpResponse> transport;
+        private Duration timeout;
+
+        public Builder url(String url) {
+            this.url = url;
+            return this;
+        }
+
+        public Builder name(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public Builder signer(Signer<HttpRequest, ?> signer) {
+            this.signer = signer;
+            return this;
+        }
+
+        public Builder transport(ClientTransport<HttpRequest, HttpResponse> transport) {
+            this.transport = transport;
+            return this;
+        }
+
+        public Builder timeout(Duration timeout) {
+            this.timeout = timeout;
+            return this;
+        }
+
+        public HttpMcpProxy build() {
+            if (url == null || url.isEmpty()) {
+                throw new IllegalArgumentException("URL must be provided");
+            }
+            return new HttpMcpProxy(this);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public CompletableFuture<JsonRpcResponse> rpc(JsonRpcRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
Don't run this asynchronously, its gonna make debugging a pain. Just run synchronously and return a completed future.